CMakeLists.txt

```diff
@@ -502,13 +502,15 @@ add_subdirectory(${GTEST_ROOT})
 find_package(GTest REQUIRED)
 
 # cudnn detection
-if(USE_CUDNN AND USE_CUDA)
```

Comment:
This `AND USE_CUDA` should not have been removed inside this PR. It assumes ` cmake_dependent_option(USE_CUDNN "Build with cudnn support" ON "USE_CUDA" OFF)` from https://github.com/apache/incubator-mxnet/pull/17018 is used. Unfortunately our tests don't test `USE_CUDA=0` on a machine with CUDNN available, so the passed.

@yajiedesign I suggest either to either merge https://github.com/apache/incubator-mxnet/pull/17018 quickly or I'll create another PR to add back the `AND USE_CUDA`